es/es-analysis-common/src/main/java/org/elasticsearch/analysis/common/LowerCaseTokenizerFactory.java

```diff
@@ -20,13 +20,15 @@
 package org.elasticsearch.analysis.common;
 
 import org.apache.lucene.analysis.Tokenizer;
-import org.apache.lucene.analysis.core.LowerCaseTokenizer;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.env.Environment;
 import org.elasticsearch.index.IndexSettings;
 import org.elasticsearch.index.analysis.AbstractTokenizerFactory;
 import org.elasticsearch.index.analysis.MultiTermAwareComponent;
 
+@Deprecated
+// Elastic set it to deprecated and will remove it, LowerCaseTokenizer is removed in Lucene 8.0.0,
+// a LowerCaseFilter should be used instead
```

Comment:
Can we use proper javadoc syntax here pls? Also I think doc should be above any annotation.